### PR TITLE
[MBL-982] Support Project Fragment's Flagging Property

### PR DESCRIPTION
**Pull request by scottkicks**
# 📲 What

Adds support for the GraphQL field:
- flagging

Also adds an output signal to `ProjectPageViewModel` that we can use in the ProjectPageViewController when we're ready.

# 🤔 Why

This new property will help tell us if a user has already flagged a project. Allowing us to keep them from reporting again.

# 🛠 How

Added this property to the `FetchProjectById` & `FetchProjectBySlug` fragments.
Rather than add this to the base fetchproject query, we can add it only to these sub queries so that we only access this property as needed.

```
...
flagging {
   id
   kind
}
...
```

# ✅ Acceptance criteria

- [x] Project page and Dashboard loads and no missing data. User can pledge/back/edit pledge/view comments/view updates, etc. No new functionality added.

### CODE REVIEW

File: Library/ViewModels/ProjectPageViewModel.swift [hunk 1]
```diff
@@ -129,6 +129,9 @@ public protocol ProjectPageViewModelOutputs {
   /// Emits `[ImageViewElement]` when the project has campaign data to download for an image row as soon as the urls are available.
   var prefetchImageURLsOnFirstLoad: Signal<[ImageViewElement], Never> { get }
 
+  /// Emits a `Bool`when a project is flagged.
```
Comment (msadoon) [inline Library/ViewModels/ProjectPageViewModel.swift:132]: space between ` and when

File: KsApi/models/graphql/adapters/Project+FetchProjectQueryData.swift
Comment (msadoon) [inline KsApi/models/graphql/adapters/Project+FetchProjectQueryData.swift:1]: Double check the tests for this file there might be a test to write.

File: KsApi/models/graphql/adapters/Project+ProjectFragment.swift
Comment (msadoon) [inline KsApi/models/graphql/adapters/Project+ProjectFragment.swift:1]: Might be a test we can write in this file as well.

File: Library/ViewModels/ProjectPageViewModel.swift [hunk 2]
```diff
@@ -129,6 +129,9 @@ public protocol ProjectPageViewModelOutputs {
...
+  var projectFlagged: Signal<Bool, Never> { get }
```
Comment (msadoon) [inline Library/ViewModels/ProjectPageViewModel.swift:133]: Probably could write a test for this new output.

**Review by msadoon — APPROVED** [review]
Approved, but check there aren't tests we can write for this new property (noted below).

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**